### PR TITLE
Miscellaneous code cleanup

### PR DESCRIPTION
**Pull request by basil**
I have been collecting a bunch of changes in a branch that were too minor to submit individually. This branch has gotten large enough that I think I should just post the diff for review. The changes are all very minor so I doubt there is a need for significant testing, but just in case I successfully ran `InjectedTest` on `text-finder` with these changes in `jenkinsci/bom`.

### CODE REVIEW

File: src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 1]
```diff
@@ -109,7 +108,7 @@ public PluginCompatTester(PluginCompatTesterConfig config) {
         runner = new ExternalMavenRunner(config.getExternalMaven());
     }
 
-    public PluginCompatReport testPlugins() throws IOException, PomExecutionException {
+    public PluginCompatReport testPlugins() throws IOException {
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/PluginCompatTester.java:111]: Exception declared but never thrown.

File: src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 2]
```diff
@@ -427,8 +426,7 @@ private TestExecutionResult testPluginAgainst(
             MavenRunner.Config mconfig,
             PomData pomData,
             PluginCompatTesterHooks pcth)
-            throws PluginSourcesUnavailableException, PomExecutionException, IOException,
-                    PomTransformationException {
+            throws PluginSourcesUnavailableException, PomExecutionException, IOException {
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/PluginCompatTester.java:429]: Exception declared but never thrown.

File: src/main/java/org/jenkins/tools/test/model/MavenPom.java [hunk 3]
```diff
@@ -414,7 +414,7 @@ private void writeDocument(final File target, final Document doc) throws IOExcep
         }
     }
 
-    private Charset getSafeCharset(final Document doc) {
+    private static Charset getSafeCharset(final Document doc) {
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/model/MavenPom.java:417]: Method could be static.

File: src/main/java/org/jenkins/tools/test/model/PluginCompatTesterConfig.java [hunk 4]
```diff
@@ -65,27 +66,27 @@ public class PluginCompatTesterConfig {
     private File war;
 
     /** A Java HOME to be used for running tests in plugins. */
-    @CheckForNull private File testJDKHome = null;
+    @CheckForNull private File testJDKHome;
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/model/PluginCompatTesterConfig.java:69]: Here and below, no need to initialize to null when that is the default value.

File: src/main/java/org/jenkins/tools/test/model/PluginCompatTesterConfig.java [hunk 5]
```diff
@@ -255,7 +256,7 @@ public Map<String, String> retrieveMavenProperties() throws IOException {
                     }
                 }
             } else {
-                throw new IOException(
+                throw new FileNotFoundException(
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/model/PluginCompatTesterConfig.java:259]: Throwing a more specific exception type.

File: src/main/java/org/jenkins/tools/test/util/ExecutedTestNamesSolver.java [hunk 6]
```diff
@@ -158,7 +158,7 @@ private static Predicate<Path> isReport(String type) {
         return file -> file.getFileName().toString().endsWith(String.format("%s-reports", type));
     }
 
-    private boolean containsFailure(NodeList nodeList) {
+    private static boolean containsFailure(NodeList nodeList) {
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/util/ExecutedTestNamesSolver.java:161]: Method could be static.

File: src/test/java/org/jenkins/tools/test/PluginCompatTesterTest.java [hunk 7]
```diff
@@ -49,21 +43,6 @@
...
-
-    @Rule public TemporaryFolder testFolder = new TemporaryFolder();
-
-    @Before
-    public void setUp() throws Exception {
-        File file = Paths.get(REPORT_FILE).toFile();
-        if (file.exists()) {
-            FileUtils.deleteQuietly(file);
-        }
-    }
```
Comment (basil) [inline src/test/java/org/jenkins/tools/test/PluginCompatTesterTest.java:65]: Dead code after the `@Ignored` or irrelevant tests were deleted.

File: src/test/java/org/jenkins/tools/test/hook/WarningsNGExecutionHookTest.java [hunk 8]
```diff
@@ -47,7 +47,7 @@ public void testCheckMethod() {
     }
 
     @Test
-    public void testAction() throws Exception {
+    public void testAction() {
```
Comment (basil) [inline src/test/java/org/jenkins/tools/test/hook/WarningsNGExecutionHookTest.java:50]: Exception declared but never thrown.

File: src/test/java/org/jenkins/tools/test/hook/WarningsNGExecutionHookTest.java [hunk 9]
```diff
@@ -66,7 +66,7 @@ public void testAction() throws Exception {
     }
 
     @Test
-    public void testType() throws Exception {
+    public void testType() {
```
Comment (basil) [inline src/test/java/org/jenkins/tools/test/hook/WarningsNGExecutionHookTest.java:69]: Exception declared but never thrown.

File: src/test/resources/org/jenkins/tools/test/jenkins-bom.xml [hunk 10]
```diff
@@ -1,430 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (basil) [inline src/test/resources/org/jenkins/tools/test/jenkins-bom.xml:1]: File no longer used after the inored or irrelevant tests were deleted.

File: src/test/resources/org/jenkins/tools/test/m2-settings.xml [hunk 11]
```diff
@@ -1,302 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (basil) [inline src/test/resources/org/jenkins/tools/test/m2-settings.xml:1]: File no longer used after the inored or irrelevant tests were deleted.

**Review by jtnord — APPROVED** [review]
looks ok pending downstream build feedback

**Pull request merged**